internal/component/discovery/discovery.go

```diff
@@ -145,14 +171,25 @@ func (c *Component) Run(ctx context.Context) error {
 			c.discMut.Lock()
 			disc := c.latestDisc
 			c.discMut.Unlock()
-			go c.runDiscovery(newCtx, disc)
+			wg.Add(1)
+			go func() {
+				c.runDiscovery(newCtx, disc)
+				c.sdMetrics.Unregister()
```

Comment:
This feels like it should break if we create the discovering multiple time in the Update.